lib/internal/modules/ts/typescript.js

```diff
@@ -0,0 +1,35 @@
+'use strict';
+const { transformSync } = require('internal/deps/swc/wasm-web');
+
+let DECODER;
+function stringify(body) {
+  if (typeof body === 'string') { return body; }
+  const { TextDecoder } = require('internal/encoding');
+  DECODER ??= new TextDecoder();
+  return DECODER.decode(body);
+}
```

Comment:
Note: this was taken from https://github.com/nodejs/node/blob/77710251e10982f847d2d3dfcf5d920b0a03a539/lib/internal/modules/esm/translators.js#L133-L144, we should probably move it to some shared module instead of duplicating.